dev/requirements.txt

```diff
@@ -37,6 +37,7 @@ numpydoc
 jinja2<3.0.0
 sphinx<3.1.0
 sphinx-plotly-directive
+sphinx-copybutton
```

Comment:
Just out of curiosity: Maybe do we need set an upper bound?

Reply:
Currently `sphinx-copybutton` latest version is `0.5.2`, so let's set the upper bound to `0.5.3`.

Reply:
Sounds good. Thanks!

Reply:
On my second thought on this, I think it would be better to remove upper bound for taking an advantage of latest version.

I submitted follow-up PR here: https://github.com/apache/spark/pull/43861.

We can set a upper bound again if CI will not working with the specific version in the future.